src/Schema/SQLiteSchemaManager.php

```diff
@@ -151,6 +151,16 @@ static function (array $a, array $b): int {
             $idx['primary']    = false;
             $idx['non_unique'] = ! $tableIndex['unique'];
 
+            if ($tableIndex['partial'] === 1) {
+                $idx['where'] = $this->connection->fetchOne(
+                    <<<'SQL'
+                    SELECT SUBSTR(sql, INSTR(sql, 'WHERE') + 6)
```

Comment:
```suggestion
                    SELECT SUBSTR(sql, INSTR(sql, 'WHERE') + LENGTH('WHERE '))
```

Question: Is this query somehow affected by the collation of a database so that an all uppercase WHERE would end up in a wrong result?

Reply:
Maybe i do not understand correctly, but if we are referring to the output of sql_master table it's documented here: https://www.sqlite.org/schematab.html#interpretation_of_the_schema_table it mentions it does some spaces conversion and keyword conversion. So I did some testing and if the partial is created with a lowercase where:
```sqlite
CREATE UNIQUE INDEX simple_partial_index ON person (id, name) where (id IS NULL)
```
instead of 
```sqlite
CREATE UNIQUE INDEX simple_partial_index ON person (id, name) WHERE (id IS NULL)
```

it would output the lowercase where, since it was created that way.
```sqlite
SELECT SUBSTR(sql, INSTR(lower(sql), 'where') + LENGTH('where '))  FROM sqlite_master WHERE type = 'index'` AND name = (?)
```
would cover both cases. Since we are only checking in sqlite_master if we are dealing with a partial index.

I can modify the implementation but do we need an additional test which proves this? I can add it if we want it

Reply:
The reason why I stuck to uppercase is because dbal itself uses the where in uppercases, but it can be used in a context where the table has been created outside of dbal, which may slightly differ.

Reply:
I have added new commits which should resolve this